ArcGISRuntimeSDKQt_CppSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.cpp

```diff
@@ -0,0 +1,226 @@
+  }
+  else
+  {
+    connect(m_offlineMapTask, &OfflineMapTask::createDefaultDownloadPreplannedOfflineMapParametersCompleted, this, [this, path] (QUuid ,const DownloadPreplannedOfflineMapParameters &parameters)
+    {
+
+      m_params = DownloadPreplannedOfflineMapParameters(parameters);
+      m_params.setUpdateMode(PreplannedUpdateMode::NoUpdates);
+
+      m_preplannedMapJob = m_offlineMapTask->downloadPreplannedOfflineMap(m_params, path);
+      connect(m_preplannedMapJob, &DownloadPreplannedOfflineMapJob::jobDone, this, &DownloadPreplannedMap::onDownloadPreplannedMapJobCompleted);
+      m_preplannedMapJob->start();
+    });
+
+    m_offlineMapTask->createDefaultDownloadPreplannedOfflineMapParameters(m_offlineMapTask->preplannedMapAreaList()->at(index));
+  }
+}
+
+void DownloadPreplannedMap::checkIfMapExists(const int index)
+{
+  if (!m_offlineMapTask)
+    return;
+
+  if (m_offlineMapTask->preplannedMapAreaList()->isEmpty())
+    return;
+
+  if (m_offlineMapTask->preplannedMapAreaList()->at(index)->loadStatus() != LoadStatus::Loaded)
+    return;
+
+  QString path = QString(m_tempPath.path() + "/" + m_offlineMapTask->preplannedMapAreaList()->at(index)->portalItem()->title() + ".mmpk");
+  QDir tempDir(path);
+
+  if (tempDir.exists())
+  {
+    m_mapExists = true;
+    emit mapExistsChanged();
+  }
+  else
+  {
+    m_mapExists = false;
+    emit mapExistsChanged();
+  }
+
+  if (m_viewingOnlineMaps)
+    m_mapView->setViewpointGeometry(m_offlineMapTask->preplannedMapAreaList()->at(index)->areaOfInterest());
+}
+
+void DownloadPreplannedMap::showOnlineMap(const int index)
+{
+  m_viewingOnlineMaps = true;
+  emit viewingOnlineMapsChanged();
+  m_mapView->setMap(m_map);
+  m_offlineMapTask->preplannedMapAreaList()->at(index)->areaOfInterest();
+  m_mapView->setViewpointGeometry(m_offlineMapTask->preplannedMapAreaList()->at(index)->areaOfInterest());
+  m_graphicsOverlay->setVisible(true);
+}
+
+void DownloadPreplannedMap::onDownloadPreplannedMapJobCompleted()
+{
+  if (m_preplannedMapJob->jobStatus() != JobStatus::Succeeded)
+    return;
+
+  m_mapView->setMap(m_preplannedMapJob->result()->offlineMap());
+  m_offlineMapTask->disconnect();
+  m_busy = false;
+  emit busyChanged();
+  m_mapExists = true;
+  emit mapExistsChanged();
+  m_viewingOnlineMaps = false;
+  emit viewingOnlineMapsChanged();
+  m_graphicsOverlay->setVisible(false);
+}
+
+void DownloadPreplannedMap::loadPreplannedMapAreas()
+{
+  m_preplannedList = m_offlineMapTask->preplannedMapAreaList();
+  m_busy = false;
+  emit preplannedListChanged();
+  emit busyChanged();
+
+  for (PreplannedMapArea* mapArea : *m_offlineMapTask->preplannedMapAreaList())
+  {
+    connect(mapArea, &PreplannedMapArea::doneLoading, this, [this, mapArea] (Error e)
+    {
+      if (!e.isEmpty())
+      {
+        qDebug() << e.message() << " - " << e.additionalMessage();
+        return;
+      }
+
+      Graphic* areaOfInterest = new Graphic(mapArea->areaOfInterest(), m_graphicParent.get());
```

Comment:
I don't see anywhere that we actually delete the `m_graphicParent` so might be simpler just to remove that and use `this` instead?